src/tribler/core/utilities/async_group/async_group.py

```diff
@@ -56,6 +61,10 @@ async def cancel(self) -> List[Future]:
 
         return active
 
+    @property
+    def cancelled(self):
+        return self._cancelled
+
     def _done_callback(self, future: Future):
         self._futures.remove(future)
```

Comment:
It should be safe to use `remove` here, but it is probably a bit safer to use `discard` instead, as an official example for `asyncio.create_task` [suggests doing in done callback](https://docs.python.org/3/library/asyncio-task.html#asyncio.create_task).

Reply:
Implemented